src/iotjs_context.h

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2015-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotjs_env.h"
+#include "iotjs_external_context.h"
+
+struct iotjs_context_t {
+  // iotjs_environment_t struct
+  iotjs_environment_t *current_env;
+
+  // Initialized
+  bool initialized;
+};
+
+#ifdef IOTJS_ENABLE_EXTERNAL_CONTEXT
+
+#define IOTJS_GET_CURRENT_CONTEXT() (iotjs_port_get_current_context())
+
+#define IOTJS_DEFINE_CURRENT_CONTEXT()                     \
+  struct iotjs_context_t *iotjs_current_external_context = \
+      IOTJS_GET_CURRENT_CONTEXT()
```

Comment:
Does this approach actually help / improve performance? In JerryScript, a similar PR is still pending (https://github.com/jerryscript-project/jerryscript/pull/2470) as its results were inconclusive at best.